server/src/comment/service/comment.service.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { CommentRepository } from '../repository/comment.repository';
+import { WriteCommentRequestDto } from '../dto/request/write-comment.dto';
+import { FeedRepository } from '../../feed/repository/feed.repository';
+import { UserRepository } from '../../user/repository/user.repository';
+import { Payload } from '../../common/guard/jwt.guard';
+import { RemoveCommentRequestDto } from '../dto/request/remove-comment.dto';
+import { EditCommentRequestDto } from '../dto/request/edit-comment.dto';
+
+@Injectable()
+export class CommentService {
+  constructor(
+    private readonly commentRepository: CommentRepository,
+    private readonly feedRepository: FeedRepository,
+    private readonly userRepository: UserRepository,
+  ) {}
+
+  async commentCheck(userInformation: Payload, commentId: number) {
+    const commentObj = await this.commentRepository.findOne({
+      where: {
+        id: commentId,
+      },
+      relations: ['user'],
+    });
+
+    if (!commentObj) {
+      throw new NotFoundException('존재하지 않는 댓글입니다.');
+    }
+
+    if (userInformation.id !== commentObj.user.id) {
+      throw new UnauthorizedException('본인이 작성한 댓글이 아닙니다.');
+    }
+  }
+
+  async create(userInformation: Payload, commentDto: WriteCommentRequestDto) {
+    const feed = await this.feedRepository.findOneBy({ id: commentDto.feedId });
+    if (!feed) {
+      throw new NotFoundException('존재하지 않는 게시글입니다.');
+    }
+
+    const user = await this.userRepository.findOneBy({
+      id: userInformation.id,
+    });
+    if (!user) {
+      throw new NotFoundException('존재하지 않는 유저입니다.');
+    }
+
+    await this.commentRepository.save({
+      comment: commentDto.comment,
+      date: new Date(Date.now()),
+      feed: { id: commentDto.feedId },
+      user: { id: user.id },
+    });
+  }
```

Comment:
```suggestion
    await this.commentRepository.save({
      comment: commentDto.comment,
      date: new Date(Date.now()),
      feed: feed,
      user: user,
    });
  }
```
P4: 이건 추측인데요, 어짜피 이미 feed와 user를 load한 상태라면 id 값이 아닌 엔티티 인스턴스를 바로 넣어주는게 더 좋지 않을까? 라고 생각했습니다.

현재 방식은 단순히 id를 담고있는 object이기에 내부적으로 `new Feed()`, `new User()` 를 수행해서 실행되지 않을까 라는 생각에 써봤습니다.

Reply:
동작 원리가 어떻게 돌아갈지는 모르겠지만, 객체로 탐색하는게 ORM에 더 적합한듯 하여 바꾸겠습니다.